woocommerce/woocommerce
# Add pre-publish sidebar

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR adds a prepublish sidebar to the product editor.

![Screenshot 2024-02-05 at 16 27 52](https://github.com/woocommerce/woocommerce/assets/1314156/9e8f934a-56a0-4f8e-b599-cff6341db718)

Closes #43704.

**ACs**

```[tasklist]
### Tasks
- [ ] Create the sidebar/modal and add it to the right side of the product editor AkVNGImLgSqCObTQ3idVn7-fi-3359_345477
- [ ] When the user clicks `Add` in the top right corner, a panel slides out from the right edge of the screen. It looks and works exactly like the pre-publish sidebar in the post/site editor.
- [ ] Add the `Add` button to the sidebar to persist the unsaved changes AkVNGImLgSqCObTQ3idVn7-fi-3359_345477
- [ ] Add the `Cancel` button to the sidebar to hide/close it AkVNGImLgSqCObTQ3idVn7-fi-3359_345477
- [ ] Add the title section of the sidebar AkVNGImLgSqCObTQ3idVn7-fi-3359_345477
```



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure the New product editor and the Prepublish sidebar (product-pre-publish-modal) are enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`.
2. Go to Products > Add New
3. Add a product `Name` and press `Add`.
4. The new prepublish sidebar should be visible.
5. Verify that the `Cancel` button closes the sidebar.
6. Press the `Add` button inside the sidebar and verify it works correctly.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
